standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -7535,6 +7523,38 @@ public List<Partition> get_partitions_by_names(final String dbName, final String
     return ret;
   }
 
+  /**
+   * Creates an instance of property manager based on the (declared) namespace.
+   * @param ns the namespace
+   * @return the manager instance
+   * @throws TException
+   */
+  private PropertyManager getPropertyManager(String ns) throws TException {
+    PropertyStore propertyStore = getMS().getPropertyStore();
+    PropertyManager mgr = PropertyManager.create(ns, propertyStore);
+    return mgr;
+  }
+  @Override
+  public PropertyGetResponse get_properties(PropertyGetRequest req) throws TException {
+    PropertyManager mgr = getPropertyManager(req.getNameSpace());
+    Map<String, PropertyMap> selected = mgr.selectProperties(req.getMapPrefix(), req.getMapPredicate(), req.getMapSelection());
```

Comment:
nit: seems like we can make `selectProperties` static in `PropertyManager`, other methods as well

Reply:
I don't see how; the store, the dirty maps logic, the JEXL engine are members of that instance. I must be missing the point.